gateway/gateway.go

```diff
@@ -551,15 +553,18 @@ func (gateway *HandleT) webRequestRouter() {
 func (gateway *HandleT) printStats() {
 	for {
 		time.Sleep(10 * time.Second)
-		logger.Info("Gateway Recv/Ack ", gateway.recvCount, gateway.ackCount)
+		recvCount := atomic.LoadUint64(&gateway.recvCount)
+		ackCount := atomic.LoadUint64(&gateway.ackCount)
+		logger.Info("Gateway Recv/Ack ", recvCount, ackCount)
 	}
 }
 
 func (gateway *HandleT) stat(wrappedFunc func(http.ResponseWriter, *http.Request)) func(http.ResponseWriter, *http.Request) {
 	return func(w http.ResponseWriter, r *http.Request) {
-		gateway.latencyStat.Start()
+		latencyStat := gateway.stats.NewStat("gateway.response_time", stats.TimerType)
```

Comment:
Good catch. So far the latency metrics are messed up then.

Reply:
Yes